api/src/main/java/com/codeforcommunity/dto/team/InviteUsersRequest.java

```diff
@@ -0,0 +1,37 @@
+package com.codeforcommunity.dto.team;
+
+import com.codeforcommunity.dto.ApiDto;
+import com.codeforcommunity.exceptions.HandledException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class InviteUsersRequest extends ApiDto {
+
+  private List<InviteContact> users;
+
+  public InviteUsersRequest(List<InviteContact> users) {
+    this.users = users;
+  }
+
+  public InviteUsersRequest() {}
+
+  public List<InviteContact> getUsers() {
+    return users;
+  }
+
+  public void setUsers(List<InviteContact> users) {
+    this.users = users;
+  }
+
+  @Override
+  public List<String> validateFields(String fieldPrefix) throws HandledException {
+    String fieldName = fieldPrefix + "invite_user_request.";
+    List<String> fields = new ArrayList<>();
+
+    if (users == null) {
+      fields.add(fieldName + "users");
+    }
+    // If we want to validate fields here: users.forEach(user -> user.validateFields(fieldPrefix));
```

Comment:
We do want to validate fields and I already gave you the code block to do that. Lets just add that code instead of a comment

Reply:
I uncommented this for now. See my previous response for this. Validating each of the users causes a problem with this requiirement in the API spec:
"If one of the email addresses is invalid or the user is already on the team that invite will not be send out, the other ones will be and a 200 OK response is returned"
I have made a different ticket to resolve this.

Reply:
This is different. When we're validating DTOs we're only checking that the JSON is the right _shape_. That means we're checking that each user here has the fields we're expecting and they are not malformed.

What the API is referring to are logical failures which happen when the request we were sent had the right fields, but because of the state of our database (users being on the same team) and/or the execution of our Emailer (email addresses being invalid) the call is invalid. These are checks that happen in the processor because we can't immediately tell from the JSON itself, we need other information.

The API is a little ambiguous about malformed emails (which get checked in the DTO) vs undeliverable emails (which isn't detected until the processor implementation) but I think it's safe to assume that we are okay to return a 400 here in the DTO if any of the emails are malformed.

So, we should have that validateFields check on each of the users as I described above and in the implementation we just have to check if a user is already on the team, and not send them an email if they are.